public/js/upload.js

```diff
@@ -115,10 +170,14 @@ document.addEventListener('DOMContentLoaded', () => {
                 throw new Error(data.error || 'Conversion failed');
             }
 
+            // Store the filename from the response
+            currentFlowFilename = data.fileName;
```

Comment:
can save this file name to a hidden input so that you can the get value on deploy